Source/WebCore/style/StyleFontSizeFunctions.cpp

```diff
@@ -183,5 +184,19 @@ int legacyFontSizeForPixelSize(int pixelFontSize, bool shouldUseFixedDefaultSize
     return findNearestLegacyFontSize<float>(pixelFontSize, fontSizeFactors, mediumSize);
 }
 
+float adjustedFontSize(float size, float sizeAdjust, float xHeight)
+{
+    float aspectValue = xHeight / size;
+    return size * ( sizeAdjust / aspectValue );
+}
+
+float adjustedFontSize(float size, float sizeAdjust, const FontMetrics& metrics)
+{
+    if (!metrics.hasXHeight())
+        return size;
```

Comment:
Under which situations would the metrics not have an x height?

instead of saying the x height may not exist, can we just guarantee at setting-time that if it wouldn't exist, we synthesize a value (maybe 60% of font size)?